VideoLocker/src/main/java/org/edx/mobile/util/IOUtils.java

```diff
@@ -0,0 +1,21 @@
+package org.edx.mobile.util;
+
+import android.support.annotation.NonNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+
+import okio.Okio;
+
+public class IOUtils {
+    @NonNull
+    public static String toString(@NonNull InputStream in, @NonNull Charset charset) throws IOException {
+        return Okio.buffer(Okio.source(in)).readString(charset);
```

Comment:
The javadoc on the `defaultCharset()` method in Marshmallow (and from the start) is:

> Returns the system's default charset. This is determined during VM startup, and will not change thereafter. On Android, the default charset is UTF-8.

It looks like this has been removed in the Android N javadoc though, as the documentation on the website now says this:

> Returns the default charset of this Java virtual machine.

> The default charset is determined during virtual-machine startup and typically depends upon the locale and charset of the underlying operating system.

Maybe Android N is making this configurable?

I concede your point that the caller should be providing the charset from the context.